engine/access/rpc/backend/backend_transactions.go

```
@@ -767,6 +769,11 @@ func (b *backendTransactions) getTransactionResultByIndexFromAnyExeNode(
 		}
 	}
 	defer logAnyError()
+
+	if len(execNodes) == 0 {
+		errs = multierror.Append(errs, errors.New("zero execution nodes provided"))
```

this should just return the error directly